## Stocks Exploratory Data Analysis
Exploratory Data Analysis (EDA) is a crucial step in data science projects. It helps in understanding the underlying patterns and relationships in the data.


## Load the dataset 

Importing the libraries and loading the dataset below which was already stored in a .csv file. The data is collected through an API, for this project I am using the polygon API https://polygon.io/ and storing the data for 5 Tech stocks of US Stock Market 

In [2]:
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
df_Microsoft=pd.read_csv(r'C:\Users\ajodh\Downloads\ctc_pyhton_ML\ML_Algos_AJ\StockAnalysisMLProject\Raw_Data\Microsoft_Data.csv')
column_names=["volume","volume weighted average price","open","close","high","low","timestamp","number_of_transactions","Date"]
df_Microsoft.columns=column_names # Renaming the columns
df_Microsoft['Date']=pd.to_datetime(df_Microsoft['Date'])

In [9]:
df_Microsoft.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 493 entries, 0 to 492
Data columns (total 10 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   volume                         493 non-null    float64       
 1   volume weighted average price  493 non-null    float64       
 2   open                           493 non-null    float64       
 3   close                          493 non-null    float64       
 4   high                           493 non-null    float64       
 5   low                            493 non-null    float64       
 6   timestamp                      493 non-null    int64         
 7   number_of_transactions         493 non-null    int64         
 8   Date                           493 non-null    datetime64[ns]
 9   year                           493 non-null    int64         
dtypes: datetime64[ns](1), float64(6), int64(3)
memory usage: 38.6 KB


From the output, we can see that the dataset has 493 rows and 9 columns. All columns have non-null values.

## Data Cleaning 

In [8]:
print(df_Microsoft.isnull().sum()) #checking for Null values

volume                           0
volume weighted average price    0
open                             0
close                            0
high                             0
low                              0
timestamp                        0
number_of_transactions           0
Date                             0
year                             0
dtype: int64
